pandas/plotting/_matplotlib/core.py

```diff
@@ -120,7 +121,77 @@ def __init__(
 
         self.sort_columns = sort_columns
 
-        self.subplots = subplots
+        if isinstance(subplots, bool):
+            self.subplots = (
+                subplots
+            )  # type: Optional[Union[bool, Sequence[Sequence[Union[str, int]]]]]
+        elif isinstance(subplots, Iterable):
+            supported_kinds = (
+                "line",
+                "bar",
+                "barh",
+                "hist",
+                "kde",
+                "density",
+                "area",
+                "pie",
+            )
+            if self._kind not in supported_kinds:
+                raise ValueError(
+                    "When subplots is an iterable, kind must be "
+                    f"one of {', '.join(supported_kinds)}. Got {self._kind}."
+                )
+
+            if any(
+                not isinstance(group, Iterable) or isinstance(group, str)
+                for group in subplots
+            ):
+                raise ValueError(
+                    "When subplots is an iterable, each entry "
+                    "should be a list/tuple of column names."
+                )
+
+            cols_in_groups = [col for group in subplots for col in group]
+            duplicates = {
+                col for (col, cnt) in Counter(cols_in_groups).items() if cnt > 1
```

Comment:
do we need the counter, can you compare the length of a set with the len of list

Reply:
Using a counter allows printing the duplicates. let me know what you prefer

Reply:
fair enough, if that's a requirement.